### PR TITLE
Refactor to add CSI driver deployment

### PR DESCRIPTION
**Pull request by darkowlzz**
This combines CSI and non-CSI deployments. All the CSI resource configs are in separate files with suffix `_csi.yaml`, and all the files have conditionals on `csi.enable` value.

The main README.md contains the same old instructions. README-CSI.md contains the CSI specific instructions.

### CODE REVIEW

File: templates/daemonset.yaml [hunk 1]
```diff
@@ -24,52 +45,76 @@ spec:
...
+        #   httpGet:
+        #     path: /v1/health
+        #     port: api
+        # readinessProbe:
+        #   initialDelaySeconds: 5
+        #   httpGet:
+        #     path: /v1/health
+        #     port: api
+        # resources:
+# {{ toYaml .Values.resources | indent 12 }}
```
Comment (darkowlzz) [inline templates/daemonset.yaml]: Will add this back later. Removed it while testing.

File: templates/NOTES.txt [hunk 2]
```diff
@@ -1,10 +1,13 @@
...
+
 1. Please run the following commands now to configure the link between Kubernetes and StorageOS(*):
 
 ClusterIP=$(kubectl get svc/{{ .Values.service.name }} --namespace {{ .Values.service.namespace }} -o custom-columns=IP:spec.clusterIP --no-headers=true)
 ApiAddress=$(echo -n "tcp://$ClusterIP:{{ .Values.service.externalPort }}" | base64)
 kubectl patch secret/{{ .Values.api.secretName }} --namespace {{ .Values.api.secretNamespace }} --patch "{\"data\": {\"apiAddress\": \"$ApiAddress\"}}"
 
 *) Unfortunately this is needed until the Kubernetes apiserver can use kube-dns to resolve service 
-   dns names like {{ .Values.service.name }}.{{ .Values.namespace }}.svc to the correct ClusterIP.
+   dns names like {{ .Values.service.name }}.{{ .Values.service.namespace }}.svc to the correct ClusterIP.
```
Comment (croomes) [inline templates/NOTES.txt]: I think this conflicts with https://github.com/storageos/helm-chart/pull/12?

**Comment by Arau** [comment]
Fantastic! Thank you for the information. It is much clear now. 

If we don't use the storageos svc as csi uses the socket. Do we actually need that svc for any other functionality, or it can be removed?

**Comment by darkowlzz** [comment]
> If we don't use the storageos svc as csi uses the socket. Do we actually need that svc for any other functionality, or it can be removed?

I'm not aware of any usage of that service, but since there's no harm in keeping it, I left it there.
Maybe the service could be used by the storageos cli to use the same service endpoint, instead of pointing to node IP.

**Pull request merged**